apps/mitt-konto/src/MittKonto/Main.purs

```diff
@@ -87,27 +89,41 @@ loadingIndicator Loading =
           ]
       }
 
-showLoadingIndicator :: SetState -> Number -> (Aff Unit -> Effect Unit)
-showLoadingIndicator setState delay =
-   \aff -> do
-     let setLoading l = liftEffect do
-           setState \s -> s { loading { login = l } }
-     Aff.launchAff_ do
-       -- the trick in here is that we want to show the loading indicator
-       -- only after the action takes some noticable time
-       Aff.bracket
-         (Aff.forkAff do
-           -- the forked action runs without blocking the thread
-           -- but returns a reference that allows to cancel it
-           Aff.delay $ Aff.Milliseconds delay
-           -- after a short sleep we enable the indicator
-           setLoading $ Just Loading)
-         (\loading -> do
-           -- when we are done we can cancel the loading indicator
-           -- so that it won't get enabled if we've finished early
-           Aff.killFiber (error "we're done") loading
-           setLoading Nothing)
-         (\loading -> aff)
+-- | Allows to run the asynchronous action while showing the loading indicator
+--   and handling the result.
+runAffLoading
+  :: forall a.
+     SetState
+  -> (Either Error a -> Effect Unit) -- ^ result handler
+  -> Aff a -- ^ asynchronous action
+  -> Effect Unit
+runAffLoading setState handler action =
+   Aff.launchAff_ do
+     loadingFiber <- Aff.forkAff do
+       -- In this thread we wait a bit and then switch the spinner on
+       -- This prevents flickering if the action completes instantly
+       Aff.delay $ Aff.Milliseconds 100.0
+       setLoading $ Just Loading
+     actionFiber <- Aff.forkAff do
+       -- In the meanwhile we run the action
+       result <- Aff.try action
+       -- then we give result to the handler (swallowing errors thrown by it)
+       Aff.apathize $ liftEffect $ handler result
+     timeoutFiber <- Aff.forkAff do
+       -- in yet another thread we sleep we are counting down till timeout
+       Aff.delay $ Aff.Milliseconds (30.0 * 1000.0)
```

Comment:
Let's have the timeout in a `timeoutSeconds` binding in a `let` on top of the function